CMakeLists.txt

```diff
@@ -64,31 +60,31 @@ include(check_packages)
 include(swig)
 include(coveralls)
 
-# add PaddlePaddle version
-if(DEFINED ENV{PADDLE_VERSION})
-    add_definitions(-DPADDLE_VERSION=\"$ENV{PADDLE_VERSION}\")
+# Set PaddlePaddle version to Git tag name or Git commit ID.
+find_package(Git REQUIRED)
+execute_process(
+  COMMAND ${GIT_EXECUTABLE} describe --tags --exact-match --abbrev=0
+  WORKING_DIRECTORY ${PROJ_ROOT}
+  OUTPUT_VARIABLE GIT_TAG_NAME
+  RESULT_VARIABLE GIT_RESULT
+  ERROR_QUIET OUTPUT_STRIP_TRAILING_WHITESPACE)
+if(NOT ${GIT_RESULT})
+  set(PADDLE_VERSION ${GIT_TAG_NAME})
```

Comment:
@wangkuiyi I see, yes it will display most recent tag.
I think here we need to discuss what is the purpose of `version`.
From my point of view, I think `version` should reflect what is currently building (current git HEAD). So if currently tag `latest-tag` is checked out as current HEAD, version should be `latest-tag`, otherwise, I think version should be `head` or `latest-tag.head` to reflect that current code is based on `latest-tag` but there are some changes.